tests/test_data_client.py

```diff
@@ -118,3 +121,75 @@ async def test_status_handling(
 
     with expectation:
         await hacs.data_client.get_repositories("integration")
+
+
+GOOD_COMMON_DATA = {
+    "description": "abc",
+    "etag_repository": "blah",
+    "full_name": "blah",
+    "last_commit": "abc",
+    "last_fetched": 0,
+    "last_updated": "blah",
+    "manifest": {},
+}
+
+GOOD_INTEGRATION_DATA = {
+    "description": "abc",
+    "domain": "abc",
+    "etag_repository": "blah",
+    "full_name": "blah",
+    "last_commit": "abc",
+    "last_fetched": 0,
+    "last_updated": "blah",
+    "manifest": {},
+    "manifest_name": "abc",
+}
+
+
+def without(d: dict, key: str) -> dict:
+    """Return a copy of d without key."""
+    d = dict(d)
+    d.pop(key)
+    return d
+
+
+@pytest.mark.parametrize(
+    ("category", "data"),
+    [
+        ("appdaemon", without(GOOD_COMMON_DATA, "description")),
+        ("integration", without(GOOD_INTEGRATION_DATA, "description")),
+        ("plugin", without(GOOD_COMMON_DATA, "description")),
+        ("python_script", without(GOOD_COMMON_DATA, "description")),
+        ("template", without(GOOD_COMMON_DATA, "description")),
+        ("theme", without(GOOD_COMMON_DATA, "description")),
+    ]
+)
+async def test_discard_invalid_repo_data(
```

Comment:
This test shows that the correct validation flag is set when updating data, but I think there should also be a test of the data client itself.